tapfreighter/wallet.go

```diff
@@ -152,6 +174,27 @@ func (s *CoinSelect) ListEligibleCoins(ctx context.Context,
 	return s.coinLister.ListEligibleCoins(ctx, constraints)
 }
 
+// LeaseCoin leases/locks/reserves a coin for the given lease owner
+// until the given expiry. This is used to prevent multiple concurrent
+// coin selection attempts from selecting the same coin.
+//
+// NOTE: The coin lock must be held when calling this function.
+func (s *CoinSelect) LeaseCoin(ctx context.Context, utxoOutpoint wire.OutPoint,
+	leaseOwner []byte, expiry time.Time) error {
+
+	return s.coinLister.LeaseCoin(ctx, utxoOutpoint, leaseOwner, expiry)
```

Comment:
Not related to this line of code, but wanting to jot this down: I think we should make sure that once an outgoing package has been written to disk, with the transfer transaction broadcast (but not yet confirmed), that UTXO shouldn't be available for leasing. Otherwise, if confirmations take a while, then we end up double spending ourselves. 

With the way things are in `btcwallet`, once a mined transaction is inserted, then the lease is also removed: https://github.com/btcsuite/btcwallet/blob/07be54bc22cf4dd2dd16a9788d2fc88952c438ae/wtxmgr/tx.go#L462-L471